typescript/sdk/src/contracts.ts

```diff
@@ -53,30 +53,42 @@ function getFactory(
   return factories[key];
 }
 
-export function filterAddresses(
-  addresses: HyperlaneAddresses<any>,
+export function filterAddressesMap(
+  addressesMap: HyperlaneAddressesMap<any>,
   factories: HyperlaneFactories,
-): HyperlaneAddresses<any> {
-  return pick(addresses, Object.keys(factories));
+): HyperlaneAddressesMap<typeof factories> {
+  // Filter out addresses that we do not have factories for
```

Comment:
I don't have full context here, so you might want somebody else to actually review this. IIUC, this is to allow to pass in a superset of addresses and still have attachContracts return the expected contracts shape. Would it be possible to not do this implicitly, but explicitly, i.e. have the caller specify the filtering explicitly?

Reply:
Doing this in a future PR